File: app/src/main/java/com/mnj/java_solutions/CanPlaceFlowersLC605.java
package com.mnj.java_solutions;

public class CanPlaceFlowersLC605 {

    void canPlaceFlower() {

        int plantSoFar = 0;

        int[] flowerBed = {1, 0, 0, 0, 1};

        int plant = 2;

        for (int i = 0; i < flowerBed.length; i++) {

            if (flowerBed[i] == 0) {

                int prev = (i == 0 || flowerBed[i - 1] == 0 ? 0 : 1);
                int next = (i == flowerBed.length - 1 || flowerBed[i + 1] == 0 ? 0 : 1);

                if (prev == 0 && next == 0) {
                    flowerBed[i] = 1;
                    plantSoFar++;
                }
            }
        }

        if (plantSoFar == plant)
            System.out.println("==>> Can plant flowers");
        else
            System.out.println("==>> Can NOT plant flowers");
    }
}

File: app/src/main/java/com/mnj/java_solutions/CanPlaceFlowersLC605.kt
package com.mnj.java_solutions

class CanPlaceFlowersLC605 {
    fun canPlaceFlower() {
        var plantSoFar = 0
        val flowerBed = intArrayOf(1, 0, 0, 0, 1)
        val plant = 2
        for (i in flowerBed.indices) {
            if (flowerBed[i] == 0) {
                val prev = if (i == 0 || flowerBed[i - 1] == 0) 0 else 1
                val next = if (i == flowerBed.size - 1 || flowerBed[i + 1] == 0) 0 else 1
                if (prev == 0 && next == 0) {
                    flowerBed[i] = 1
                    plantSoFar++
                }
            }
        }
        if (plantSoFar == plant) println("==>> Can plant flowers") else println("==>> Can NOT plant flowers")
    }
}
